libc/string/string_lenchr.c

```diff
@@ -52,105 +54,306 @@ TEST(string_len, ascii)
-	TEST_ASSERT_EQUAL_INT(sz - 1, strnlen(bigstr, sz - 1));
-	TEST_ASSERT_EQUAL_INT(sz, strnlen(bigstr, sz));
-	TEST_ASSERT_EQUAL_INT(sz, strnlen(bigstr, sz + 1));
+	free((void *)asciiStr);
+}
+
+
+TEST(string_spn, not_ascii)
+{
+	unsigned char charSet[BUFF_SIZE] = { 0 };
+	int len, i;
+
+	/* Checking  out of ASCII bytes */
+	for (i = sizeof(charSet); i < sizeof(charSet) * 2 - 1; i++) {
+		charSet[i - sizeof(charSet)] = i;
+	}
+
+	len = sizeof(charSet) - 1;
+	TEST_ASSERT_EQUAL_INT(len, strcspn((const char *)charSet, ""));
+	TEST_ASSERT_EQUAL_INT(len, strspn((const char *)charSet, (const char *)charSet));
+}
+
+
+TEST(string_spn, big)
+{
+	char bigstr[PATH_MAX] = { 0 };
+	int len;
+
+	/*
+	 * The length of the string is not restricted,
+	 * so we test one of the bigger values, which may be used
+	 * with remembering the last element must be a null term
+	 * zero to collect the data as the string from
+	 * declared space, that's why we memset one place less than its size
+	 */
+
+	len = sizeof(bigstr) - 1;
+	memset(bigstr, 'a', len);
+
+	TEST_ASSERT_EQUAL_INT(len, strcspn(bigstr, ""));
+	TEST_ASSERT_EQUAL_INT(0, strcspn(bigstr, "a"));
+	TEST_ASSERT_EQUAL_INT(len, strcspn(bigstr, "b"));
+	TEST_ASSERT_EQUAL_INT(0, strcspn(bigstr, "ab"));
+
+	TEST_ASSERT_EQUAL_INT(len, strspn(bigstr, "a"));
+	TEST_ASSERT_EQUAL_INT(0, strspn(bigstr, ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn(bigstr, "b"));
+	TEST_ASSERT_EQUAL_INT(len, strspn(bigstr, "ab"));
+
+	TEST_ASSERT_EQUAL_INT(len, strcspn(bigstr, ""));
+	TEST_ASSERT_EQUAL_INT(0, strcspn(bigstr, "a"));
+	TEST_ASSERT_EQUAL_INT(len, strcspn(bigstr, "b"));
+	TEST_ASSERT_EQUAL_INT(0, strcspn(bigstr, "ab"));
+
+	TEST_ASSERT_EQUAL_INT(0, strspn(bigstr, ""));
+	TEST_ASSERT_EQUAL_INT(len, strspn(bigstr, "a"));
+	TEST_ASSERT_EQUAL_INT(0, strspn(bigstr, "b"));
+	TEST_ASSERT_EQUAL_INT(len, strspn(bigstr, "ab"));
+}
+
+
+TEST(string_spn, empty_args)
+{
+	TEST_ASSERT_EQUAL_INT(0, strcspn("", "abc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn("", "abc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("", ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn("", ""));
+
+	TEST_ASSERT_EQUAL_INT(3, strcspn("abc", ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn("abc", ""));
+}
+
+
+TEST(string_spn, first_byte)
+{
+	/* This case is intended to check the ability of stopping at first byte */
+	TEST_ASSERT_EQUAL_INT(1, strcspn("abc", "bc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn("abc", "bc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("abc", "ab"));
+	TEST_ASSERT_EQUAL_INT(2, strspn("abc", "ab"));
+
+	TEST_ASSERT_EQUAL_INT(4, strcspn("aaaabc", "bc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn("aaaabc", "bc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("aaaabc", "ab"));
+	TEST_ASSERT_EQUAL_INT(5, strspn("aaaabc", "ab"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("aaaabc", "a"));
+	TEST_ASSERT_EQUAL_INT(4, strspn("aaaabc", "a"));
+}
+
+
+TEST(string_spn, mixed_order)
+{
+	int i;
+	char *reversStr,
+		*testStr;
+
+	testStr = testdata_createCharStr(BUFF_SIZE);
+	reversStr = malloc(BUFF_SIZE);
```

Comment:
I'm considering whether it is reasonable to use malloc with the current value of BUFF_SIZE.
```suggestion
	char *reversStr[BUFF_SIZE],
		*testStr;

	testStr = testdata_createCharStr(BUFF_SIZE);
```